Blog Post for Autobuild docs for Pull Requests (#72)
readthedocs/blog: building-docs-in-pull-requests.rst

  we synchronize the external version for that pull request and trigger a new documentation build with the latest changes.

- **Deleting External Versions:** Whenever the pull request is closed or merged,
  we delete the external Version associated with that pull request.

- **Warning Banner for pull request documentation:** While building documentation for pull requests
  we add a warning banner at the top of those documentations to let the users know that
  this documentation was generated for pull requests and is not the main documentation for the project.

We send build status reports to status API of the provider (e.g. GitHub).
When a build is triggered for a pull request we send build pending notification with the build URL
and after the build has finished we send success notification if the build succeeded without any error
or failure notification if the build failed.
By going to the build URL provided in the status report users can view the build steps
and also see the documentation generated by that build.

**Currently, we only support GitHub and planning to extend to GitLab and BitBucket**

Getting Started
===============

> Build documentation in pull requests is currently on Beta testing and only supports Github repositories.

Reviewer: ```suggestion
Building documentation for pull requests is currently in Beta testing and only supports Github repositories.
```

> .. post:: July 14, 2019

Reviewer 2: This need to be updated. Let's plan for tomorrow?

```suggestion
.. post:: Aug 14, 2019
```

   :tags: gsoc, pr-builder, continuous-documentation, feature
   :author: Saadmk11
   :location: DHA

> Autobuild Documentation in Pull Requests

Reviewer 2: ```suggestion
GSOC 2019: Autobuild Documentation in Pull Requests
```

I think this is good to mention 👍

> ========================================

Reviewer 2: (Need to update this too, to match :D

- **Deleting External Versions:** Whenever the pull request is closed or merged,
  we delete the external Version associated with that pull request.

- **Warning Banner for pull request documentation:** While building documentation for pull requests
  we add a warning banner at the top of those documentations to let the users know that
  this documentation was generated for pull requests and is not the main documentation for the project.

We send build status reports to status API of the provider (e.g. GitHub).
When a build is triggered for a pull request we send build pending notification with the build URL
and after the build has finished we send success notification if the build succeeded without any error
or failure notification if the build failed.
By going to the build URL provided in the status report users can view the build steps
and also see the documentation generated by that build.

**Currently, we only support GitHub and planning to extend to GitLab and BitBucket**

Getting Started
===============

Building documentation for pull requests is currently in Beta testing and only supports Github repositories.

> If you want to dive in and enable this feature for your project you can `email us <mailto:support@readthedocs.org>`_.

Reviewer 2: We should mention that people should include a project URL that they want to enable.

Future Improvements
===================

We are planing to extend this feature to other platforms such as GitLab and BitBucket.
We are also planning to make this feature more customizable through our configaration file (``.readthedocs.yaml``).
If you have any improvements or features in mind for building documentation in pull requests we would love to know about it.

> Please feel free to `let us know <mailto:team@readthedocs.org>`_.

Reviewer 2: I'd say to file an issue on GitHub here.
